src/main/java/org/jenkinsci/plugins/gitclient/verifier/KnownHostsFileVerifier.java

```diff
@@ -32,51 +32,29 @@ public AbstractCliGitHostKeyVerifier forCliGit(TaskListener listener) {
 
     @Override
     public AbstractJGitHostKeyVerifier forJGit(TaskListener listener) {
-        KnownHosts knownHosts;
-        try {
-            if (Files.exists(getKnownHostsFile().toPath())) {
-                knownHosts = new KnownHosts(getKnownHostsFile());
-            } else {
+        Path knowHostPath = getKnownHostsFile().toPath();
+        if (Files.notExists(knowHostPath)) {
+            try {
                 logHint(listener);
-                knownHosts = new KnownHosts();
+                Files.createDirectories(knowHostPath.getParent());
+                Files.createFile(knowHostPath);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, e, () -> "Could not load known hosts.");
             }
-        } catch (IOException e) {
-            LOGGER.log(Level.WARNING, e, () -> "Could not load known hosts.");
-            knownHosts = new KnownHosts();
         }
-        return new KnownHostsFileJGitHostKeyVerifier(listener, knownHosts);
+        return new KnownHostsFileJGitHostKeyVerifier(listener);
     }
 
-    public class KnownHostsFileJGitHostKeyVerifier extends AbstractJGitHostKeyVerifier {
-
-        private final TaskListener listener;
+    public static class KnownHostsFileJGitHostKeyVerifier extends AbstractJGitHostKeyVerifier {
 
-        public KnownHostsFileJGitHostKeyVerifier(TaskListener listener, KnownHosts knownHosts) {
-            super(knownHosts);
-            this.listener = listener;
-        }
-
-        @Override
-        public String[] getServerHostKeyAlgorithms(Connection connection) throws IOException {
-            return getPreferredServerHostkeyAlgorithmOrder(connection);
+        public KnownHostsFileJGitHostKeyVerifier(TaskListener listener) {
+            super(listener);
         }
 
         @Override
-        public boolean verifyServerHostKey(
-                String hostname, int port, String serverHostKeyAlgorithm, byte[] serverHostKey) throws Exception {
-            listener.getLogger()
-                    .printf(
-                            "Verifying host key for %s using %s %n",
-                            hostname, getKnownHostsFile().toPath());
-            LOGGER.log(Level.FINEST, "Verifying {0}:{1} in known hosts file {2} with host key {3} {4}", new Object[] {
-                hostname,
-                port,
-                SshHostKeyVerificationStrategy.KNOWN_HOSTS_DEFAULT,
-                serverHostKeyAlgorithm,
-                Base64.getEncoder().encodeToString(serverHostKey)
-            });
-            return verifyServerHostKey(
-                    listener, getKnownHosts(), hostname, port, serverHostKeyAlgorithm, serverHostKey);
+        public OpenSshConfigFile.HostEntry customizeHostEntry(OpenSshConfigFile.HostEntry hostEntry) {
+            hostEntry.setValue(SshConstants.STRICT_HOST_KEY_CHECKING, SshConstants.YES);
+            return hostEntry;
```

Comment:
confused why we are working at a low level when higher primatives exist that can seemingly hide this complexity?

Reply:
https://github.com/jenkinsci/git-client-plugin/pull/1127#discussion_r1566668664